src/goto-symex/json.cpp

```diff
@@ -0,0 +1,386 @@
+#include <util/message.h>
+#include <goto-symex/goto_trace.h>
+#include <util/language.h>
+#include <langapi/language_util.h>
+#include <nlohmann/json.hpp>
+#include <filesystem>
+#include <fstream>
+#include <regex>
+
+using json = nlohmann::json;
+
+namespace {
+    // Helper function to check if string starts with prefix, as CI system prefers starts_with
+    bool starts_with(const std::string& str, const std::string& prefix) {
+        return str.size() >= prefix.size() && 
+               str.compare(0, prefix.size(), prefix) == 0;
+    }
+
+    // Helper function to read file lines
+    std::vector<std::string> read_file_lines(const std::string& filename) {
+        std::vector<std::string> lines;
+        std::ifstream input(filename);
+        std::string line;
+        
+        while(std::getline(input, line)) {
+            lines.push_back(std::move(line));
+        }
+        
+        return lines;
+    }
+
+    std::string type_id_to_string(type2t::type_ids id) {
+        switch(id) {
+            case type2t::bool_id: return "bool";
+            case type2t::empty_id: return "empty";
+            case type2t::symbol_id: return "symbol";
+            case type2t::struct_id: return "struct";
+            case type2t::union_id: return "union";
+            case type2t::pointer_id: return "pointer";
+            case type2t::array_id: return "array";
+            case type2t::code_id: return "code";
+            case type2t::fixedbv_id: return "fixedbv";
+            case type2t::unsignedbv_id: return "unsigned";
+            case type2t::signedbv_id: return "signed";
+            default: return fmt::format("unknown({})", static_cast<int>(id));
+        }
+    }
+
+    std::set<std::string> find_included_headers(const std::string& file_path, std::set<std::string>& processed_files) {
+        if (processed_files.find(file_path) != processed_files.end() || 
+            starts_with(std::filesystem::path(file_path).string(), "/usr/")) {
```

Comment:
I think you are the same check a few times now. Consider adding a function and documenting why the /usr is the only prefix to check